Use 256 byte alignment to avoid breakages starting with cublas 12.9.1.4.

dimvar opened the pull request:
The following tests fail without this change:

//xla/backends/gpu/runtime:gpublas_lt_matmul_thunk_test_nvgpu_any
//xla/service/gpu:gpu_compiler_test_nvgpu_any
//xla/service/gpu/transforms:cublas_gemm_rewriter_test_nvgpu_any
//xla/service/gpu/transforms:gemm_rewriter_test_nvgpu_any

dimvar committed:
Use 256 byte alignment to avoid breakages starting with cublas 12.9.1.4.

derdrdirk commented:
This change breaks the following tests:

xla/backends/gpu/codegen:dynamic_slice_fusion_test_2gpu
xla/service/gpu/tests:calling_convention.hlo.test
xla/service/gpu/tests:gpu_alignment_test_gpu
xla/service/gpu/tests:gpu_alignment_test_h100
xla/service/gpu/tests:gpu_noalias_test_h100
xla/service/gpu/tests:kernel_reuse.hlo.test

Once fixed I'll try to merge the PR.

dimvar committed:
Changes alignment to 256 in the following tests:
//xla/service/gpu/tests:calling_convention.hlo.test
//xla/service/gpu/tests:gpu_alignment_test_h100
//xla/service/gpu/tests:gpu_noalias_test_h100
//xla/service/gpu/tests:kernel_reuse.hlo.test

dimvar commented:
I don't see target //xla/backends/gpu/codegen:dynamic_slice_fusion_test_2gpu. Is this google-internal only? I see xla/backends/gpu/codegen:dynamic_slice_fusion_test and it passes. I fixed all other failures.

dimvar committed:
Fixed dynamic_slice_fusion_rewriter_test to use 256-byte alignment.
